lib/clear.js

```diff
@@ -0,0 +1,169 @@
+import {
+    client,
+    categoriesService,
+    ordersService,
+    stateService,
+    customersService,
+    shippingMethodService,
+    zonesService,
+    productsService
+} from './services'
+
+const deleteCategories = () => {
+    client.execute({
+        uri: categoriesService.build(),
+        method: 'GET',
+    }).then(result => {
+            return Promise.all(
+                    result.body.results.map((element) => {
+                        if (!element.parent) {
+                            return client.execute({
+                                uri: categoriesService.byId(element.id).withVersion(element.version).build(),
+                                method: 'DELETE',
+                            })
+                        }
+                        
+                    })
+                ).then(console.log('All categories are succesfully deleted'))
+                .catch(err => console.log(err))
+                }
+        ).catch(err => console.log(err));
+}
+
+const deleteOrders = () => {
+    client.execute({
+        uri: ordersService.perPage(500).build(),
```

Comment:
same here,  client.process(...)  is your friend for iterating through everything  https://commercetools.github.io/nodejs/sdk/api/sdkClient.html#processrequest-processfn-options 

500 will get you into trouble (you get one huge JSON response in one very slow API call and you have no idea how big it will get, you could even get an error message back from the API.

An alternative when deleting is to use GraphQL to fetch _only_ the ids and versions to be able to brute-force delete. You don't really need anything else from the API payload and that way you can safely fetch 500 or 1000 in one call.

Reply:
I will try to make it work with `client.process(...)` and about GraphQL I have no experience with it yet but I see that it has it's advantages here so I will use it here as well.

Reply:
I suggest you leave out GraphQL for now - one thing at a time :-).  If you know the JS SDK really well after this that's a good goal.